apps/backend/src/reconciliation/pos-reconciliation.service.ts

```diff
@@ -281,17 +306,28 @@ export class PosReconciliationService {
               },
               deposit: {
                 ...deposit,
+                reconciled_on:
+                  process.env.RUNTIME_ENV === 'production'
+                    ? currentDate
+                    : parse(dateToFind, 'yyyy-MM-dd', new Date()),
                 heuristic_match_round: posHeuristicRound,
                 status: MatchStatus.MATCH,
                 timestamp: deposit.timestamp,
               },
             });
             // Pull it out of the dictionary
             if (deposits.length) {
-              depositsWithAmount[dateToFind] = deposits;
+              if (posHeuristicRound === PosHeuristicRound.THREE) {
+                depositsWithAmount[dayBeforeDateToFind] = deposits;
```

Comment:
Can we assume that it's the dayBeforeDateTofFind just because its round 3? First instinct is yes, but if you can think of other scenarios, I'm curious

Reply:
If it is round three, then we set the deposits dictionary to this: 

deposits = depositsWithAmount[`${dayBeforeDateToFind}-${paymentMethod}`];

I think we have to assume the deposits would be dayBeforeDateToFind, because if they are not, then the reconciliation would be incorrectly matching something that should have matched on rounds 1 or 2... (but maybe I am missing something?)